mirdata/annotations.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+"""
+    TODO all docs
+    """
+
+import numpy as np
+from dataclasses import dataclass
+
+
+# Check consistency
+
+
+def check_types(attributes_names, attributes, expected_type, expected_dtype):
+    for name, attribute, exp_type, int_type in zip(
+        attributes_names, attributes, expected_type, expected_dtype
+    ):
+        if attribute is not None:
+            if not isinstance(attribute, exp_type):
+                raise TypeError(
+                    f"{name} should be a {exp_type}, but is a {type(attribute)} instead"
+                )
+            if not np.asarray(attribute).size == 0:  # check list/ndarray not empty
+                if all(isinstance(type(x), int_type) for x in attribute):
```

Comment:
eek! just found this bug it should be `isinstance(s, int_type)`. When its fixed, there are a lotttt of broken loaders. fixing them now 😱